nixos/modules/profiles/appliance.nix

```diff
@@ -0,0 +1,25 @@
+# This profile sets up a sytem for appliance usage. An appliance is installed
+# as an image, cannot be re-built, has no Nix available, and is generally not
+# meant for interactive use.
+
+{ lib, modulesPath, ... }:
+
+{
+
+  # Appliances are always "minimal".
+  imports = [
+    "${modulesPath}/profiles/minimal.nix"
+  ];
+
+  # The system cannot be rebuilt.
+  nix.enable = false;
+  system.rebuildable = false;
```

Comment:
This should be factored out into `virtual-appliance.nix` which can then import this file.
Hardware appliances need an upgrade mechanism, and that would probably involve `switch`-ing.